internal/pipe/pipe.go

```diff
@@ -86,11 +88,21 @@ func (c *Cmd) WPipe() (int, io.WriteCloser, error) {
 	return fd, w, nil
 }
 
-func (c *Cmd) registerPipe(local, remote *os.File) int {
-	// From https://pkg.go.dev/os/exec#Cmd.ExtraFiles: "entry i becomes file
-	// descriptor 3+i".
-	fd := 3 + len(c.Cmd.ExtraFiles)
-	c.Cmd.ExtraFiles = append(c.Cmd.ExtraFiles, remote)
+func (c *Cmd) registerPipe(local, remote *os.File) uintptr {
+	var fd uintptr
+	// Cmd.ExtraFiles is not supported on Windows.
+	if runtime.GOOS == "windows" {
+		fd = remote.Fd()
+		if c.Cmd.SysProcAttr == nil {
+			c.Cmd.SysProcAttr = &syscall.SysProcAttr{}
+		}
+		c.Cmd.SysProcAttr.AdditionalInheritedHandles = append(c.Cmd.SysProcAttr.AdditionalInheritedHandles, syscall.Handle(fd))
```

Comment:
It looks like this doesn't compile, at least with the latest Go versions and on Linux. Could you confirm that it does compile on Windows?

If it does compile on Windows, would adding a separate pipe_windows.go help here?